### PR TITLE
CLDR-15319 build: CodeQL Analysis

### PR DESCRIPTION
**Pull request by srl295**
CLDR-15319

While doing dependabot, GitHub suggested CodeQL analysis which is in a similar tab. This will report any issues to us.
I updated the suggested action script to use our own maven build.

- [ ] This PR completes the ticket.

### CODE REVIEW

**Review by btangmu — DISMISSED** [review]
that will be interesting

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+    - name: Build Java
+      if: matrix.language == 'java'
+      run: >
+        mvn -s .github/workflows/mvn-settings.xml -B compile --file tools/pom.xml
+        -DskipTests=true -Dskip.npm -Dskip.installnodenpm
```
Comment (srl295) [inline .github/workflows/codeql-analysis.yml:77]: skip lines here speed up the maven build…

**Comment by srl295** [comment]
Hm! Lots of failures. Maybe we shold look at them before proceeding.

**Comment by btangmu** [comment]
>

**Comment by btangmu** [comment]
sorry I accidentally clicked on a Close button

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java [hunk 2]
```diff
@@ -1001,8 +1001,8 @@ public static void showCoverage(Anchors anchors, Matcher matcher, Set<String> lo
                     + "<th style='text-align:left'>" + "Locales" + "</th>"
                     + "</tr>"
                 );
-            int totalCount = 0;
-            int totalLocaleCount = 0;
+            long totalCount = 0;
+            long totalLocaleCount = 0;
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/ShowLocaleCoverage.java:1005]: seemed to be an easy way to quell the errs…

**Comment by srl295** [comment]
@btangmu  @macchiati  CodeQL no longer complains here.

**Comment by srl295** [comment]
> Lots of security warnings...

Yes, And some of them are fixed in this PR.  Like to merge this and then work on fixing others.

**Review by btangmu — APPROVED** [review]
is there an easy off-switch if we get inundated by warnings?

**Comment by srl295** [comment]
> is there an easy off-switch if we get inundated by warnings?

We can just revert the workflow.

**Pull request merged**